libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts

```diff
@@ -1050,4 +1067,86 @@ export const buildTigerSpecificFunctions = (
             });
         });
     },
+    getAllCSPDirectives: async (): Promise<Array<ICSPDirective>> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getAllEntitiesCspDirectives({});
+                return result.data?.data || [];
+            });
+        } catch (error) {
+            return [];
+        }
+    },
+    getCSPDirective: async (directiveId: string): Promise<ICSPDirective> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getEntityCspDirectives({ id: directiveId });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            throw convertApiError(error);
+        }
+    },
+    createCSPDirective: async (requestData: ICSPDirective): Promise<ICSPDirective> => {
+        try {
+            const jsonApiCspDirectiveInDocument: JsonApiCspDirectiveInDocument = {
+                data: {
+                    id: requestData.id,
+                    type: JsonApiCspDirectiveInTypeEnum.CSP_DIRECTIVE,
+                    attributes: requestData.attributes,
+                },
+            };
+
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.createEntityCspDirectives({
+                    jsonApiCspDirectiveInDocument,
+                });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            if (error.response?.status === 400) {
+                const message = error?.response?.data?.detail
+                    ? error?.response?.data?.detail
+                    : "Server error";
+                throw new UnexpectedError(message, error);
+            }
+            throw convertApiError(error);
+        }
+    },
+    updateCSPDirective: async (directiveId: string, requestData: ICSPDirective): Promise<ICSPDirective> => {
+        try {
+            const jsonApiCspDirectiveInDocument: JsonApiCspDirectiveInDocument = {
+                data: {
+                    id: requestData.id,
+                    type: JsonApiCspDirectiveInTypeEnum.CSP_DIRECTIVE,
+                    attributes: requestData.attributes,
+                },
+            };
+
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.updateEntityCspDirectives({
+                    id: directiveId,
+                    jsonApiCspDirectiveInDocument,
+                });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            if (error.response?.status === 400) {
+                const message = error?.response?.data?.detail
+                    ? error?.response?.data?.detail
+                    : "Server error";
+                throw new UnexpectedError(message, error);
+            }
+            throw convertApiError(error);
```

Comment:
Can we delete lines 1134 ~ 1139 and keep just this line only? Logic from lines 1134 ~ 1139 is probably already contained in `convertApiError`...

Reply:
`convertApiError` did not handle the statusCode = 400 case.

Reply:
It does return `UnexpectedError` at the following place: https://github.com/gooddata/gooddata-ui-sdk/blob/2fac540cb6cae33df4239e9e92cc97e3d949bcda/libs/sdk-backend-tiger/src/utils/errorHandling.ts#L35

→ Do we really need to get `message` from the error and then pass it to UnexpectedError's constructor?